x-pack/docs/en/rest-api/security/query-api-key.asciidoc

```diff
@@ -0,0 +1,258 @@
+[role="xpack"]
+[[security-api-query-api-key]]
+=== Query API key information API
+++++
+<titleabbrev>Query API key information</titleabbrev>
+++++
+
+Retrieves information for API keys with <<query-dsl,Query DSL>>
+in a <<paginate-search-results,pagniated>> fashion.
+
+[[security-api-query-api-key-request]]
+==== {api-request-title}
+
+`GET /_security/_query/api_key`
+
+`POST /_security/_query/api_key`
+
+[[security-api-query-api-key-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the `manage_own_api_key` cluster privilege.
+
+[[security-api-query-api-key-desc]]
+==== {api-description-title}
+
+The information for the API keys created by
+<<security-api-create-api-key,create API Key>> can be retrieved using this API in
+a paginated manner and optionally filtered with a query.
+
+[[security-api-query-api-key-request-body]]
+==== {api-path-body-title}
+
+The following parameters can be specified in the body of a request and
+pertain to retrieving api keys:
+
+`query`::
+(Optional, string) A <<query-dsl,query>> to filter API keys to be returned.
+Note it only support a subset of the query types including
+<<query-dsl-match-all-query>>, <<query-dsl-bool-query>>,
+<<query-dsl-term-query>>, <<query-dsl-terms-query>>, <<query-dsl-ids-query>>,
+<<query-dsl-prefix-query>>, <<query-dsl-wildcard-query>>, <<query-dsl-range-query>>.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=from]
++
+By default, you cannot page through more than 10,000 hits using the `from` and
+`size` parameters. To page through more hits, use the
+<<search-after,`search_after`>> parameter.
+
+`size`::
+(Optional, integer) The number of hits to return. Needs to be non-negative.
++
+By default, you cannot page through more than 10,000 hits using the `from` and
+`size` parameters. To page through more hits, use the
+<<search-after,`search_after`>> parameter.
```

Comment:
This is repeated from above. Is that intentional?

Reply:
Yes it is indeed intentional. The same is repeated on the [Search API doc page](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html). The difference is that the two parameters, `from` and `size`, are further apart on the Search API page which makes the repitition more justifiable. I can consolidate them if you prefer.